backends/cuda-ref/ceed-cuda-ref-operator.c

```diff
@@ -310,7 +349,7 @@ static inline int CeedOperatorSetupInputs_Cuda(CeedInt num_input_fields, CeedQFu
         uint64_t state;
 
         CeedCallBackend(CeedVectorGetState(vec, &state));
-        if (state != impl->input_states[i] && !impl->skip_rstr_in[i]) {
+        if ((state != impl->input_states[i] || vec == in_vec) && !impl->skip_rstr_in[i]) {
```

Comment:
I'm not sure I follow this change. Does `impl->input_states` not correctly track the state of active inputs? I guess that makes sense since the user could hand over a brand new `CeedVector` as the active input and the operator has no way of knowing that?

Reply:
The input states do keep track of active input states fine. The issue is that if I'm using the same buffer for input and output e-vecs corresponding to the active inputs/outptus, then I always need to re-do the active input restriction as the data is always overwritten. Since we rarely rerun the operator with identical input data, I don't see this costing us much here

Reply:
Ok. I would say it might be worth adding a comment to that effect to clarify the purpose of `vec == in_vec` here. Something like
```c
// Always apply restriction to active inputs due to potential reuse of e-vecs
```